semver.md

```diff
@@ -9,7 +9,7 @@ Given a version number MAJOR.MINOR.PATCH, increment the:
 1. MAJOR version when you make incompatible API changes,
 1. MINOR version when you add functionality in a backwards-compatible
    manner, and
-1. PATCH version when you make backwards-compatible bug fixes.
+1. PATCH version when you make backwards-compatible changes.
```

Comment:
Under this definition, how would you distinguish MINOR from PATCH?

> MINOR version when you add functionality in a backwards-compatible

Adding functionality is also a "change" so the distinction between MINOR and PATCH is lost with the proposed changes here.

The point of PATCH is that no additional functionality is added. It is only fixes to _existing_ functionality.

Reply:
@Haacked, good catch, I can't believe I missed that.

How about "PATCH version when you make backwards-compatible changes that do not change the interface"  ?

Reply:
imo it is very clear, if you ADD functionality, minor, if you change without changing functionality patch.

If you follow the reasoning by @Haacked then there also is no distinction between MAJOR and the others, as MAJOR also entails changes

> Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backwards compatible functionality is introduced to the public API.

Mind the "new" in the above ^^^. 

Those lines at the top are basically the TLDR version of the spec. The details further down are very clear.

Reply:
How about this?

MAJOR version when you make incompatible API changes,
MINOR version when you add to the API in a backwards-compatible manner, and
PATCH version when you make any other backwards to compatible changes.

Reply:
You could very well make backward compatible changes that would merit a PATCH level release:

``` diff
-int hash(char *str);
+int hash(const char *str);
```

For all intents and purposes, this change doesn't do anything other than to openly declare that `hash` should do what it probably already doesn't do, i.e. it doesn't change the contents pointed at by `str`, and now says so clearly.

This change, however, merits a MAJOR release, as this is an *incompatible* change:

``` diff
-int hash(const char *str);
+uint64_t hash(const char *str);
```